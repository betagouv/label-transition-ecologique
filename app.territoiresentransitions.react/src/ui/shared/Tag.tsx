import classNames from 'classnames';

type TTag = {
  title: string;
  onCloseClick?: () => void;
  className?: string;
};

const Tag = ({className, title, onCloseClick}: TTag) => {
  return (
    <div
      className={classNames(
        'flex items-center px-4 py-1 text-white bg-bf500 rounded-full',
        {'pr-2': onCloseClick},
        className
      )}
    >
      <span className="mb-1 text-sm">{title}</span>
      {onCloseClick && (
        <button
          className="ml-1 rounded-full"
          onClick={evt => {
            evt.stopPropagation();
            onCloseClick();
          }}
        >
          <div className="fr-fi-close-line flex m-auto scale-75" />
        </button>
      )}
    </div>
  );
};

export default Tag;
