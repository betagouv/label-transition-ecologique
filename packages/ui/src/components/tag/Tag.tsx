import classNames from 'classnames';

type TagProps = {
  /** Libellé affiché dans l'étiquette */
  title: string;
  /** Appelée lors du clic sur le bouton "Fermer". Ne pas spécifier pour masquer le bouton. */
  onClose?: () => void;
  /** Pour désactiver les interactions */
  disabled?: boolean;
  /** Permet de surcharger les styles */
  className?: string;
  /** Change la couleur par défaut pour distinguer les étiquettes prédéfinies
   * des étiquettes utilisateur */
  isUserCreated?: boolean;
};

/** Affiche une étiquette */
export const Tag = ({
  className,
  title,
  disabled,
  onClose,
  isUserCreated,
}: TagProps) => {
  if (!title || title.length === 0) {
    return null;
  }
  const canClose = onClose && !disabled;

  return (
    <div
      className={classNames(
        'flex items-center px-3 py-0.5 text-white bg-primary rounded-full',
        { 'pr-2': canClose },
        {
          '!bg-white !border !border-gray-300 !text-gray-900': isUserCreated,
        },
        className
      )}
    >
      <span className="py-0.5 text-sm">{title}</span>
      {canClose && (
        <div
          className="ml-1 rounded-full cursor-pointer"
          onClick={(evt) => {
            evt.stopPropagation();
            onClose();
          }}
        >
          <div className="fr-fi-close-line flex m-auto scale-75" />
        </div>
      )}
    </div>
  );
};
