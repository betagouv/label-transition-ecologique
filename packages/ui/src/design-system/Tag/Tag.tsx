import {Icon} from '@design-system/Icon';
import classNames from 'classnames';

type TagProps = {
  /** Libellé affiché dans l'étiquette */
  title: string;
  /** Appelée lors du clic sur le bouton "Fermer". Ne pas spécifier pour masquer le bouton. */
  onClose?: () => void;
  /** Pour désactiver les interactions */
  disabled?: boolean;
  /** Permet de surcharger les styles */
  className?: string;
  /** Change la couleur par défaut pour distinguer les étiquettes prédéfinies
   * des étiquettes utilisateur */
  isUserCreated?: boolean;
  /** Restreint le titre à une seule ligne */
  trim?: boolean;
};

/** Affiche une étiquette */
export const Tag = ({
  className,
  title,
  disabled,
  onClose,
  isUserCreated,
  trim = false,
}: TagProps) => {
  if (!title || title.length === 0) {
    return null;
  }
  const canClose = onClose && !disabled;

  return (
    <div
      className={classNames(
        'flex items-center px-3 py-0.5 rounded-full',
        isUserCreated
          ? 'text-gray-900 bg-white border border-gray-300'
          : 'text-white bg-primary',
        {'pr-2': canClose},
        className
      )}
    >
      <span className={classNames('py-0.5 text-sm', {'line-clamp-1': trim})}>
        {title}
      </span>
      {canClose && (
        <div
          className="flex ml-1 rounded-full cursor-pointer"
          onClick={evt => {
            evt.stopPropagation();
            onClose();
          }}
        >
          <Icon icon="close-line" size="md" className="m-auto" />
        </div>
      )}
    </div>
  );
};
