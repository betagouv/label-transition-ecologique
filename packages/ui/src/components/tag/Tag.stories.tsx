import React from 'react';
import { action } from '@storybook/addon-actions';
import { Tag } from './Tag';

export default {
  component: Tag,
};

export const Default = () => (
  <div className="flex">
    <Tag title="Yolo" />
  </div>
);

export const AvecBoutonFermer = () => (
  <div className="flex">
    <Tag title="Yolo" onClose={action('onClose')} />
  </div>
);

export const CreeParUtilisateur = () => (
  <div className="flex">
    <Tag title="Yolo" isUserCreated onClose={action('onClose')} />
  </div>
);

export const StylePersonnalise = () => (
  <div className="flex">
    <Tag title="Yolo" className="!bg-error-1 font-bold" />
  </div>
);
