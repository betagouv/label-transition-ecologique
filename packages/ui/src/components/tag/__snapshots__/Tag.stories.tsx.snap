// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/tag/Tag AvecBoutonFermer smoke-test 1`] = `
<div class="flex">
  <div class="flex items-center px-3 py-0.5 rounded-full text-white bg-primary pr-2">
    <span class="py-0.5 text-sm">
      Yolo
    </span>
    <div class="ml-1 rounded-full cursor-pointer">
      <div class="fr-fi-close-line flex m-auto scale-75">
      </div>
    </div>
  </div>
</div>
`;

exports[`components/tag/Tag CreeParUtilisateur smoke-test 1`] = `
<div class="flex">
  <div class="flex items-center px-3 py-0.5 rounded-full text-gray-900 bg-white border border-gray-300 pr-2">
    <span class="py-0.5 text-sm">
      Yolo
    </span>
    <div class="ml-1 rounded-full cursor-pointer">
      <div class="fr-fi-close-line flex m-auto scale-75">
      </div>
    </div>
  </div>
</div>
`;

exports[`components/tag/Tag Default smoke-test 1`] = `
<div class="flex">
  <div class="flex items-center px-3 py-0.5 rounded-full text-white bg-primary">
    <span class="py-0.5 text-sm">
      Yolo
    </span>
  </div>
</div>
`;

exports[`components/tag/Tag StylePersonnalise smoke-test 1`] = `
<div class="flex">
  <div class="flex items-center px-3 py-0.5 rounded-full text-white bg-primary !bg-error-1 font-bold">
    <span class="py-0.5 text-sm">
      Yolo
    </span>
  </div>
</div>
`;
